packages/zendesk-adapter/src/filters/guide_arrange_paths.ts

```diff
@@ -0,0 +1,253 @@
+  CATEGORY_TRANSLATION_TYPE_NAME,
+  GUIDE_SETTINGS_TYPE_NAME,
+  USER_SEGMENT_TYPE_NAME,
+  PERMISSION_GROUP_TYPE_NAME,
+  GUIDE_LANGUAGE_SETTINGS_TYPE_NAME,
+  ZENDESK, GUIDE, BRAND_TYPE_NAME,
+} from '../constants'
+import {
+  ARTICLES_ORDER,
+  CATEGORIES_ORDER,
+  // GUIDE_ORDER_TYPES,
+  SECTIONS_ORDER,
+} from './guide_order/guide_orders_utils'
+
+const { RECORDS_PATH } = elementsUtils
+const log = logger(module)
+
+export const UNSORTED = 'unsorted'
+export const GUIDE_PATH = [ZENDESK, RECORDS_PATH, GUIDE]
+const FIRST_LEVEL_TYPES = [USER_SEGMENT_TYPE_NAME, PERMISSION_GROUP_TYPE_NAME]
+const BRAND_SECOND_LEVEL = [
+  CATEGORY_TYPE_NAME,
+  GUIDE_SETTINGS_TYPE_NAME,
+  GUIDE_LANGUAGE_SETTINGS_TYPE_NAME,
+  CATEGORIES_ORDER,
+]
+const PARENTS = [CATEGORY_TYPE_NAME, SECTION_TYPE_NAME, ARTICLE_TYPE_NAME]
+const TRANSLATIONS = [
+  CATEGORY_TRANSLATION_TYPE_NAME,
+  SECTION_TRANSLATION_TYPE_NAME,
+  ARTICLE_TRANSLATION_TYPE_NAME]
+
+const OTHER_TYPES = [...TRANSLATIONS, SECTIONS_ORDER, ARTICLES_ORDER]
+export const GUIDE_ELEMENT_DIRECTORY: Record<string, string> = {
+  [ARTICLE_TRANSLATION_TYPE_NAME]: 'translations',
+  [ARTICLE_TYPE_NAME]: 'articles',
+  [CATEGORY_TYPE_NAME]: 'categories',
+  [SECTION_TYPE_NAME]: 'sections',
+  [SECTION_TRANSLATION_TYPE_NAME]: 'translations',
+  [CATEGORY_TRANSLATION_TYPE_NAME]: 'translations',
+  [GUIDE_SETTINGS_TYPE_NAME]: 'settings',
+  [USER_SEGMENT_TYPE_NAME]: 'user_segments',
+  [PERMISSION_GROUP_TYPE_NAME]: 'permission_groups',
+  [GUIDE_LANGUAGE_SETTINGS_TYPE_NAME]: 'language_settings',
+  [CATEGORIES_ORDER]: 'categories_order',
+  [SECTIONS_ORDER]: 'sections_order',
+  [ARTICLES_ORDER]: 'articles_order',
+}
+
+/**
+ * calculates a path which is not related to a specific brand
+ */
+const pathForGlobalTypes = (instance: InstanceElement): readonly string[] | undefined =>
+  [
+    ...GUIDE_PATH,
+    GUIDE_ELEMENT_DIRECTORY[instance.elemID.typeName],
+    pathNaclCase(instance.elemID.name),
+  ]
+
+
+/**
+ * calculates a path which is related to a specific brand and does not have a parent
+ */
+const pathForBrandSpecificRootElements = (instance: InstanceElement, brandName: string | undefined)
+  : readonly string[] => {
+  if (brandName === undefined) {
+    log.error('brandName was not found for instance %s.', instance.elemID.getFullName())
+    return [
+      ...GUIDE_PATH,
+      UNSORTED,
+      GUIDE_ELEMENT_DIRECTORY[instance.elemID.typeName],
+      pathNaclCase(instance.elemID.name),
+    ]
+  }
+  const newPath = [
+    ...GUIDE_PATH,
+    'brands',
+    brandName,
+    GUIDE_ELEMENT_DIRECTORY[instance.elemID.typeName],
+    pathNaclCase(instance.elemID.name),
+  ]
+  if (instance.elemID.typeName === CATEGORY_TYPE_NAME) { // each category has a folder of its own
+    newPath.push(pathNaclCase(instance.elemID.name))
+  }
+  return newPath
+}
+
+/**
+ * calculates a path which is related to a specific brand and has a parent.
+ */
+const pathForOtherLevels = (params :{
```

Comment:
nit: when all parameters are used it's sometimes nicer to expand this
```suggestion
const pathForOtherLevels = ({
  instance,
  needTypeDirectory,
  needOwnFolder,
  parent,
} :{
```
and then you don't need `params.<>` everywhere